### ISSUE TITLE
V2 logging support

### ISSUE THREAD

**Issue by michalpristas** [elastic/elastic-agent#1489]
After #954 is in proceed based on v2 architecture with investigation of passing writer to sub-processes

compare passing `writer` and `file handle`
rotate logs without a need to update passed `file handle`
make sure there's no significant performance impact

**Comment by cmacknz** [comment]
@michalpristas does this overlap with https://github.com/elastic/elastic-agent/issues/221?

**Comment by michalpristas** [comment]
yes

**Comment by michalpristas** [comment]
closing in favor of #221
